### PR TITLE
Update cipher tool to read deployment toml secrets

### PR DESCRIPTION
**Pull request by madushadhanushka**
## Purpose
Change cipher tool to read deployment toml file secrets section and update the same toml file with encrypted data. Example deployment toml secrets section will be as follows. Field to be encrypted should be surround with square brackets.
```
[secrets]
password="[wso2carbon]"
```
Cipher tool can be run with following command to encrypt the secret fields.
```
./ciphertool.sh -Dconfigure
```

This command convert all secrete properties in deployment toml into encrypted format.

### CODE REVIEW

File: components/ciphertool/src/main/java/org/wso2/ciphertool/utils/Utils.java [hunk 1]
```diff
@@ -366,5 +382,46 @@ public static String doEncryption(Cipher cipher, String plainTextPwd) {
...
+     * Read toml file and return list of secrets
+     *
+     * @param configFilePath    file path to deployment toml
+     * @return      Map of secrets
+     */
+    public static Map<String, String> getSecreteFromConfiguration(String configFilePath) {
+        Map<String, String> context = new LinkedHashMap<>();
+        try {
+            TomlParseResult result = Toml.parse(Paths.get(configFilePath));
+            result.errors().forEach(error -> System.out.print(error.toString()));
```
Comment (jsdjayanga) [inline components/ciphertool/src/main/java/org/wso2/ciphertool/utils/Utils.java]: Why do you continue if you have errors?

File: components/ciphertool/src/main/java/org/wso2/ciphertool/CipherTool.java [hunk 2]
```diff
@@ -53,9 +57,22 @@ public static void main(String[] args) {
         Cipher cipher = KeyStoreUtil.initializeCipher();
         if (System.getProperty(Constants.CONFIGURE) != null &&
             System.getProperty(Constants.CONFIGURE).equals(Constants.TRUE)) {
-            loadXpathValuesAndPasswordDetails();
-            secureVaultConfigTokens();
-            encryptCipherTextFile(cipher);
+            Map<String, String> secretMap = Utils.getSecreteFromConfiguration(Utils.getDeploymentFilePath());
```
Comment (jsdjayanga) [inline components/ciphertool/src/main/java/org/wso2/ciphertool/CipherTool.java]: 1. Check whether the toml file is available.
2. Check why you haven't call loadXpathValuesAndPasswordDetails, secureVaultConfigTokens, encryptCipherTextFile, if there are no secrets
3. Add tests to test this section

File: features/org.wso2.ciphertool.feature/resources/bin/ciphertool.sh [hunk 3]
```diff
@@ -104,7 +104,7 @@ fi
 
 # update classpath
 CARBON_CLASSPATH=""
-for f in "$CARBON_HOME"/lib/org.wso2.ciphertool*.jar
+for f in "$CARBON_HOME"/lib/*
```
Comment (tharindu1st) [inline features/org.wso2.ciphertool.feature/resources/bin/ciphertool.sh]: make *.jar

File: components/ciphertool/src/main/java/org/wso2/ciphertool/utils/Utils.java [hunk 4]
```diff
@@ -345,6 +351,16 @@ private static String getAbsolutePathWithCarbonHome(String keyStorePath, String
...
     }
+
+    /**
+     * Get deployment toml file path
+     *
+     * @return  deployment file path
+     */
+    public static String getDeploymentFilePath() {
+        String homeFolder = System.getProperty(Constants.CARBON_HOME);
+        return Paths.get(homeFolder, Constants.CONF_DIR, Constants.DEPLOYMENT_TOML_FILE).toString();
```
Comment (tharindu1st) [inline components/ciphertool/src/main/java/org/wso2/ciphertool/utils/Utils.java]: read from system property

**Pull request merged**